crates/pypi_mapping/src/custom_pypi_mapping.rs

```diff
@@ -45,6 +47,21 @@ where
     Ok(mapping_by_name)
 }
 
+pub fn fetch_mapping_from_path<T: DeserializeOwned, P: AsRef<Path>>(path: &P) -> miette::Result<T> {
```

Comment:
From the name of the function I would expect this to return a mapping but instead it returns anything that is deserializable. Imo we can just remove the generics.